[CHANGE] Removing legacy unity preprocessors mentions for 2022.3 or older

josepmariapujol-unity opened the pull request:
### Description

This PR removes the legacy Unity preprocessors mentions for editors 2022.3 or older (e.g. 2021.x, 2020.x).

As stated in the [package.json](https://github.com/Unity-Technologies/InputSystem/blob/b60f6a76339f263fe67f939fe85cb67a72ab7b0d/Packages/com.unity.inputsystem/package.json#L5) minimum editor version supported is 2022.3.

### Testing status & QA

Compilation

### Overall Product Risks

- Complexity: Low
- Halo Effect: Low

### Comments to reviewers

_Please describe any additional information such as what to focus on, or historical info for the reviewers._

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

josepmariapujol-unity committed:
Update XboxGamepadMacOS.cs

josepmariapujol-unity committed:
Update InputSystemUIInputModule.cs

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=2279) <br/>All committers have signed the CLA.

josepmariapujol-unity committed:
Removing unity legacy preprocessors for 2022.3 or older

josepmariapujol-unity committed:
Update CHANGELOG.md for version changes

Removed mentions of legacy Unity preprocessors for older editor versions and updated the changelog for version 1.16.0.

josepmariapujol-unity commented:
Duplicate PR: https://github.com/Unity-Technologies/InputSystem/pull/2274
